api/controller/assignments.js

```diff
@@ -0,0 +1,321 @@
+        if (err) {
+            res.send({ 'error': 'An error has occurred in finding assignments' });
+        } else {
+            collection.find().toArray(function (err, items) {
+                res.send(items);
+
+            });
+        }
+    });
+};
+
+exports.launchAssignment = function (req, res) {
+    //validate
+    if (!req.params.assingid) {
+        res.status(400).send({
+            error: "Assignment id is not defined",
+            code: 22
+        });
+        return;
+    }
+
+    //find assignment by id
+    db.collection(assignmentCollection, function (err, collection) {
+        collection.findOne({ _id: new mongo.ObjectID(req.params.assingid) }, function (err, assignment) {
+            if (err) {
+                res.send({ 'error': 'An error has occurred in finding assignment' });
+            } else {
+                var assignmentData;
+                assignmentData = assignment;
+                //find all students from classrooms
+                var classrooms = assignmentData.classrooms;
+                var privateJournalIds = []; //to store private journal ids of every student
+                var students = []; //to store all students data 
+
+
+                //get students private_journal
+                common.fetchAllStudents(classrooms).then(function (stds) {
+                    //student data
+                    students = students.concat(stds);
+
+                    //getting priavte journal ids from every student id
+                    var uniqueJournalIds = [];
+                    for (var i = 0; i < stds.length; i++) {
+                        if (uniqueJournalIds.indexOf(stds[i].private_journal.toString()) == -1) {
+                            uniqueJournalIds.push(stds[i].private_journal.toString());
+                        }
+                    }
+
+                    privateJournalIds = uniqueJournalIds;
+                    var entryDoc = {};
+
+                    db.collection(journalCollection, function (err, collection) {
+                        if (err) {
+                            res.send({ "error": "An error has occurred in finding journal Entry" });
+                        } else {
+                            //find entry by private_journal and ObjectID
+                            console.log(assignmentData.assignedWork.toString());
+                            //aggregate query to get entry on the bisis of private_journal and objectId
+                            collection.aggregate([
+                                {
+                                    $match: {
+                                        '_id': new mongo.ObjectID(req.user.private_journal),
+                                        "content.objectId": assignmentData.assignedWork
+                                    }
+                                },
+                                {
+                                    $unwind: "$content"
```

Comment:
$unwind consumes higher memory. There are alternatives to unwind. Try to implement them.